File: web/src/api/partsApi.js
const API_BASE_URL = import.meta.env.VITE_API_BASE_URL

async function handleResponse(response) {
    if (!response.ok) {
        const errorBody = await response.json().catch(() => ({}))
        throw new Error(errorBody.message || 'Something went wrong.')
    }
    return response.json()
}

export async function getAllParts() {
    const response = await fetch(`${API_BASE_URL}/api/parts`)
    return handleResponse(response)
}

export async function getAvailableParts() {
    const response = await fetch(`${API_BASE_URL}/api/parts/available`)
    return handleResponse(response)
}

export async function searchParts(query) {
    const response = await fetch(`${API_BASE_URL}/api/parts/search?query=${encodeURIComponent(query)}`)
    return handleResponse(response)
}

export async function createPart({ name, category, stockQuantity, isAvailable }) {
    const response = await fetch(`${API_BASE_URL}/api/parts`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ name, category, stockQuantity, isAvailable }),
    })
    return handleResponse(response)
}

export async function updatePart(id, { name, category, stockQuantity, isAvailable }) {
    const response = await fetch(`${API_BASE_URL}/api/parts/${id}`, {
        method: 'PUT',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ name, category, stockQuantity, isAvailable }),
    })
    return handleResponse(response)
}

export async function updatePartAvailability(id, isAvailable) {
    const response = await fetch(`${API_BASE_URL}/api/parts/${id}/availability`, {
        method: 'PATCH',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ isAvailable }),
    })
    return handleResponse(response)
}

export async function updatePartStock(id, stockQuantity) {
    const response = await fetch(`${API_BASE_URL}/api/parts/${id}/stock`, {
        method: 'PATCH',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ stockQuantity }),
    })
    return handleResponse(response)
}

export async function deletePart(id) {
    const response = await fetch(`${API_BASE_URL}/api/parts/${id}`, {
        method: 'DELETE',
    })
    if (!response.ok) {
        const errorBody = await response.json().catch(() => ({}))
        throw new Error(errorBody.message || 'Failed to delete part.')
    }
}

File: application/app/src/main/java/edu/cit/ballener/lakbayayos/data/api/PartsApi.kt
package edu.cit.ballener.lakbayayos.data.api

import edu.cit.ballener.lakbayayos.data.model.Part
import edu.cit.ballener.lakbayayos.data.model.PartRequest
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.PUT
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface PartsApi {

    // Used by customers browsing/searching
    @GET("/api/parts/available")
    suspend fun getAvailableParts(): List<Part>

    @GET("/api/parts/search")
    suspend fun searchParts(@Query("query") query: String): List<Part>

    // Used by admin managing inventory
    @GET("/api/parts")
    suspend fun getAllParts(): List<Part>

    @POST("/api/parts")
    suspend fun createPart(@Body request: PartRequest): Part

    @PUT("/api/parts/{id}")
    suspend fun updatePart(@Path("id") id: Long, @Body request: PartRequest): Part

    @PATCH("/api/parts/{id}/availability")
    suspend fun updateAvailability(@Path("id") id: Long, @Body body: Map<String, Boolean>): Part

    @PATCH("/api/parts/{id}/stock")
    suspend fun updateStock(@Path("id") id: Long, @Body body: Map<String, Int>): Part
}
